### PR TITLE
Enable more warnings when compiling.

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

File: src/nucleus/tt.ml [hunk 1]
```diff
@@ -312,124 +312,6 @@ and abstract_term_ty xs ?(lvl=0) (e, t) =
   and t = abstract_ty xs ~lvl t
   in (e, t)
 
-let shift_abstraction shift_u shift_v k lvl us v =
```
Comment (andrejbauer) [inline src/nucleus/tt.ml]: I take it `shift_abstraction` was never used?

File: src/nucleus/tt.ml [hunk 2]
```diff
@@ -802,11 +684,6 @@ and print_projection xs te xts p ppf =
       (print_term ~max_level:0 xs te)
       (Name.print_ident p)
 
-and print_binder xs (x, t) ppf =
```
Comment (andrejbauer) [inline src/nucleus/tt.ml]: Is this dead code?

File: src/parser/desugar.ml [hunk 3]
```diff
@@ -25,11 +25,6 @@ let rec mk_prod ~loc ys ((t', _) as t) =
       | _ -> Syntax.Prod (ys, t), loc
     end
 
-let mk_let ~loc w c =
```
Comment (andrejbauer) [inline src/parser/desugar.ml]: Dead code?

**Comment by SkySkimmer** [comment]
All the Tt.shift functions are unused. Tt.print_binder is also unused (binders with parentheses are before my time). Desugar.mk_let was used for the computation/expression separation, and became unused with #89.

**Pull request merged**